localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker_factory.py

```diff
@@ -149,8 +157,23 @@ def get_esm_worker(self) -> EsmWorker:
                 processor=esm_processor,
             )
         else:
-            raise Exception(
-                f"Unsupported event source mapping source service {source_service}. Please upvote or create a feature request."
+            poller_holder = {"create_poller": None}
+            lambda_hooks.create_event_source_poller.run(
+                poller_holder, source_service, self.esm_config
+            )
+
+            create_poller: Optional[Callable[..., Type[Poller]]] = poller_holder.get(
+                "create_poller"
+            )
+            if not create_poller:
+                raise Exception(
+                    f"Unsupported event source mapping source service {source_service}. Please upvote or create a feature request."
+                )
+
+            poller: Type[Poller] = create_poller(
```

Comment:
```suggestion
            poller: Poller = create_poller(
```

`create_poller` is most likely the type here, but the result should be `Poller`, right?

Reply:
`create_poller` is a function used to generate the `Poller`, where we expect the result to be a child instance of the base class `Poller`.

The way a child class of a parent `Parent` is "type-hinted" in Python is via the `Type[Parent]` syntax.

From the [Python docs](https://mypy.readthedocs.io/en/latest/kinds_of_types.html#the-type-of-class-objects):
> Sometimes you want to talk about class objects that inherit from a given class. This can be spelled as type[C] (or, on Python 3.8 and lower, [typing.Type[C]](https://docs.python.org/3/library/typing.html#typing.Type)) where C is a class.

Reply:
I think you misunderstood that documentation. We are not talking about the class itself, we are talking about instances of that class. The callable returned returns an instance of that class, not the class itself, so the `type[]` is not correct here.

For your type hints, the create_poller function would look like this:
```python
def _return_value(*args, **kwargs):
    return SubclassedPoller
```

But since the create_poller method (whatever it exactly is) creates an instance of the class, it is just `poller: Poller`.